Gems/AWSCore/Code/Include/Framework/AWSApiRequestJob.h

```diff
@@ -277,22 +277,30 @@ namespace AWSCore
         virtual void DoCleanup()
         {
         }
+    };
 
+    template<class RequestTraits>
+    class AwsApiRequestJob : public AwsApiRequestJobBase<RequestTraits>
+    {
     public:
-        using OnSuccessFunction = AZStd::function<void(AwsApiRequestJob* job)>;
-        using OnFailureFunction = AZStd::function<void(AwsApiRequestJob* job)>;
+        using OnSuccessFunction = AZStd::function<void(AwsApiRequestJob<RequestTraits>* job)>;
+        using OnFailureFunction = AZStd::function<void(AwsApiRequestJob<RequestTraits>* job)>;
+        using IConfig = IAwsApiRequestJobConfig<RequestTraits>;
+
+        // To use a different allocator, extend this class and use this macro.
+        AZ_CLASS_ALLOCATOR(AwsApiRequestJob, AZ::SystemAllocator, 0);
 
         /// A specialization of AwsApiRequestJob that lets you provide functions
         /// that are called on success or failure of the request.
-        class Function : public AwsApiRequestJob
+        class Function : public AwsApiRequestJobBase<RequestTraits>
```

Comment:
Same as above: Can we use the AwsApiRequestJobType alias here instead of AwsApiRequestJobBase<RequestTraits>?

Reply:
Yes, we can do that

Reply:
With this change, I think that the outer parent of Function is AwsApiRequestJob instead of AwsApiRequestJobBaseand the alias is defined in AwsApiRequestJobBase actually. There still might be some problems because of the template though.

Reply:
This approach has changed. The inner Function derived class no longer exists, it has been merged into the base class so that the ::Create static function will still work as needed. And there is no longer any inheritance (with the inner Function class or the new class to solve that problem)